### PR TITLE
Adding vm::ref<T> support for non-iree_vm_ref_t ref types.

### PR DESCRIPTION
**Pull request by benvanik**

### CODE REVIEW

File: tools/iree-benchmark-module-main.cc [hunk 1]
```diff
@@ -437,20 +437,20 @@ class IREEBenchmark {
...
+    instance_.reset();
 
     // Tear down device last in order to get accurate statistics.
-    if (device_allocator_ && FLAG_print_statistics) {
-      IREE_IGNORE_ERROR(
-          iree_hal_allocator_statistics_fprint(stderr, device_allocator_));
+    if (!!device_allocator_ && FLAG_print_statistics) {
+      IREE_IGNORE_ERROR(iree_hal_allocator_statistics_fprint(
+          stderr, device_allocator_.get()));
     }
```
Comment (ScottTodd) [inline tools/iree-benchmark-module-main.cc:448]: Is this `!!device_allocator_` intended? Why not just `device_allocator_`? Maybe find another way to express "ref has a value"?

Comment (benvanik) [reply tools/iree-benchmark-module-main.cc:448]: for some reason the bool operator stuff doesn't work - I'll see if I can figure it out

Comment (ScottTodd) [reply tools/iree-benchmark-module-main.cc:448]: ok, thanks. (LGTM either way, just this nit)

**Review by ScottTodd — APPROVED** [review]

File: tools/iree-benchmark-module-main.cc (hunk 1 above)
Comment (benvanik) [reply tools/iree-benchmark-module-main.cc]: fixed (if gcc is happy)! 🤞

**Pull request merged**